### DISCUSSION TITLE
tkinter with atom

### DISCUSSION THREAD

**Discussion by bestevenyoung** [atom/atom#23202] [Q&A]
Hi folks,

Just a quick question from a beginner. 

I tried to make a popup window in Atom using Python (very simple). 
---
import tkinter as tk

window = tk.Tk()
---
When I run the code in IDLE, I get the expected result. An empty window pops up. But if I run it in Atom, there's no popup. Any help is appreciated. Thanks in advance.

**Comment by clivelo** [comment]
You should call `mainloop()` on your window at the end of your code.

For example:
```
import tkinter as tk

window = tk.Tk()
greeting = tk.Label(text="Hello World!")
greeting.pack()

window.mainloop()
```
